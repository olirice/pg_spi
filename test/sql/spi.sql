drop extension if exists pg_spi;
create extension pg_spi;

drop table if exists abc;
create table abc(id int primary key);

-- Execute and confirm it was inserted
select execute_and_commit('insert into abc(id) values (1)');
select * from abc;


select execute_and_rollback('insert into abc(id) values (2)');
select * from abc;
