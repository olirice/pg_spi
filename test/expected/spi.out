drop extension if exists pg_spi;
NOTICE:  extension "pg_spi" does not exist, skipping
create extension pg_spi;
drop table if exists abc;
NOTICE:  table "abc" does not exist, skipping
create table abc(id int primary key);
-- Execute and confirm it was inserted
select execute_and_commit('insert into abc(id) values (1)');
 execute_and_commit 
--------------------
                  1
(1 row)

select * from abc;
 id 
----
  1
(1 row)

select execute_and_rollback('insert into abc(id) values (2)');
ERROR:  portal snapshots (0) did not account for all active snapshots (1)
select * from abc;
 id 
----
  1
(1 row)

